RSC Publishing.js

```diff
@@ -35,52 +35,64 @@
 
 	***** END LICENSE BLOCK *****
 */
+
 function getResults(doc) {
-	/**Both search result and book ToC pages use javascript to load content, so
+	/** Both search result and book ToC pages use javascript to load content, so
 	 * this actually doesn't work as intended. Search results will work, but
 	 * will also trigger on empty result set. detectWeb for book ToC does not
 	 * work, but doWeb does,
 	 */
-	return ZU.xpath(doc, '//div[@id="all" or @id="chapterList"]\
-						//div[contains(@class,"title_text")]\
-						//a[not(contains(@href,"/database/"))]');
+	return ZU.xpath(doc, '//div[@id="all" or @id="chapterList"]//div[contains(@class,"title_text")]//a[not(contains(@href,"/database/"))]');
 }
 
 function detectWeb(doc, url) {
-	if (url.search(/\/results[?\/]/i) != -1 || url.indexOf('/ebook/') != -1  &&
-		getResults(doc).length) {
+	if (url.match(/\/results[?/]/i) != -1 || url.includes('/ebook/')
```

Comment:
@AbeJellinek not sure why you changed this from `search` to `match`, but as written, the first expression will always evaluate to true (`match` returns an array or null, never -1). Generally I've used `search` or `test` when evaluating regular expressions for T/F since they're a good bit more efficient.

Reply:
Oh no, I meant to remove the `!= -1`. I'll fix that. We have a rule in the style checker disallowing `s.indexOf(x) != -1`-style constructions, and I think using `search` that way has the same weaknesses.